backends/ebpf/psa/ebpfPsaGen.cpp

```diff
@@ -690,6 +690,31 @@ bool ConvertToEBPFControlPSA::preorder(const IR::ExternBlock* instance) {
     return false;
 }
 
+class IndexVisitor : public Inspector {
+    bool used = false;
+    const IR::PathExpression *readIndexExpr;
+ public:
+    explicit IndexVisitor(const IR::PathExpression *expr) {
+        readIndexExpr = expr;
+    }
+    bool preorder(const IR::AssignmentStatement* a) override;
+
+    bool readIndexChanged();
+};
+
+bool IndexVisitor::preorder(const IR::AssignmentStatement *a) {
+    if (auto expr = a->left->to<IR::PathExpression>()) {
+        if(expr->path->name.name == readIndexExpr->path->name.name)
```

Comment:
no space

Reply:
Also, for visitors that are used only locally I often embed them in an empty namespace to prevent them from polluting the C++ global namespace of names,